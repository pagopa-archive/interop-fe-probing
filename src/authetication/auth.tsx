import awsmobile from './aws-exports'
import { Auth, Amplify } from 'aws-amplify'

Amplify.configure(awsmobile)

async function login(username: string, password: string) {
  try {
    const user = await Auth.signIn(username, password)
    const token = user.signInUserSession.idToken.jwtToken
    const refreshToken = user.signInUserSession.refreshToken.token
    const accessToken = user.signInUserSession.accessToken.jwtToken
    sessionStorage.setItem('token', token)
    sessionStorage.setItem('refreshToken', refreshToken)
    sessionStorage.setItem('accessToken', accessToken)
    return user
  } catch (error) {
    console.log('error signing in', error)
  }
}

async function logout() {
  try {
    await Auth.signOut()
    sessionStorage.clear()
  } catch (error) {
    console.log('error signing out: ', error)
  }
}

export { login, logout }
