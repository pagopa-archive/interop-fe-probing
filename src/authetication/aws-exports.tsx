/* eslint-disable */
// WARNING: DO NOT EDIT. This file is automatically generated by AWS Amplify. It will be overwritten.

const awsmobile = {
  aws_project_region: import.meta.env.VITE_AWS_PROJECT_REGION,
  aws_cognito_identity_pool_id: null,
  aws_cognito_region: import.meta.env.VITE_AWS_COGNITO_REGION,
  aws_user_pools_id: import.meta.env.VITE_AWS_USER_POOL_ID,
  aws_user_pools_web_client_id: import.meta.env.VITE_AWS_USER_POOLS_WEB_CLIENT_ID,
  oauth: {},
  aws_cognito_username_attributes: ['EMAIL'],
  aws_cognito_social_providers: [],
  aws_cognito_signup_attributes: ['EMAIL'],
  aws_cognito_mfa_configuration: 'OFF',
  aws_cognito_mfa_types: [],
  aws_cognito_password_protection_settings: {
    passwordPolicyMinLength: 8,
    passwordPolicyCharacters: [],
  },
  aws_cognito_verification_mechanisms: ['EMAIL'],
}

export default awsmobile
