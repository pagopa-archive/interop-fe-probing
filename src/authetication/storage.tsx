const setStorage = (name: string, value: string) => {
  return new Promise((resolve, reject) => {
    sessionStorage.setItem(name, value);
    return resolve("OK");
  });
};

const deleteStorage = (name: string) => {
  return new Promise((resolve, reject) => {
    return resolve(sessionStorage.removeItem(name));
  });
};

const resetStorage = () => {
  return new Promise((resolve, reject) => {
    return resolve(sessionStorage.clear());
  });
};

const getStorage = (item: string) => {
  return new Promise((resolve, reject) => {
    return resolve(sessionStorage.getItem(item));
  });
};

export { setStorage, deleteStorage, resetStorage, getStorage };
