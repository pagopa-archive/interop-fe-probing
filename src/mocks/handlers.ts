import { rest } from 'msw'

export const handlers = [
  rest.get('http://localhost:3000/eservices', (req, res, ctx) => {
    const mockResponse = {
      content: [
        {
          id: '2',
          eserviceName: 'Anagrafica comunale',
          producerName: 'Comune di Torino',
          state: 'OFFLINE',
          versionNumber: 8,
        },
        {
          id: '1',
          eserviceName: 'Anagrafica comunale',
          producerName: 'Comune di Milano',
          state: 'ONLINE',
          versionNumber: 7,
        },
        {
          id: '3',
          eserviceName: 'Anagrafica comunale',
          producerName: 'Comune di Torino',
          state: 'ONLINE',
          versionNumber: 7,
        },
        {
          id: '10',
          eserviceName: 'Erogazione borse di studio',
          producerName:
            'A.Di.S.U.R.C. Azienda per Il Diritto allo Studio Universitario della Regione Campania',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '11',
          eserviceName: 'Funzionamento pannelli solari',
          producerName: 'Comune di Brindisi',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '4',
          eserviceName:
            'Posizione centraline ambientali con un nome davvero molto molto molto lungo',
          producerName: 'Comune di Pescara',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '5',
          eserviceName: 'Postazioni e-bike',
          producerName: 'Comune di Roma',
          state: 'OFFLINE',
          versionNumber: 1,
        },
        {
          id: '6',
          eserviceName: 'Stato colonnine di ricarica auto elettriche',
          producerName: 'Comune di Bari',
          state: 'OFFLINE',
          versionNumber: 1,
        },
        {
          id: '7',
          eserviceName: 'Stato lavori Olimpiadi 2026',
          producerName: 'Comune di Milano',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '8',
          eserviceName: 'Stato opere PGT 2030',
          producerName: 'Comune di Trento',
          state: 'n/d',
          versionNumber: 1,
        },
      ],
      offset: 0,
      limit: 10,
      totalElements: 11,
    }

    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/producers', (req, res, ctx) => {
    const mockResponse = [
      {
        label: 'producer2',
        value: 'producer2',
      },
    ]
    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/eservices/mainData/', (req, res, ctx) => {
    const mockResponse = {
      eserviceName: 'Probing test 3',
      producerName: 'Comune di Milano',
      versionNumber: '7',
      pollingFrequency: 5,
    }
    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/eservices/probingData/', (req, res, ctx) => {
    const mockResponse = {
      probingEnabled: true,
      state: 'offline',
      responseReceived: '2023-04-27T14:55:25Z',
      eserviceActive: false,
    }
    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/telemetryData/eservices/', (req, res, ctx) => {
    const mockResponse = {
      values: [
        {
          responseTime: 640,
          status: 'OK',
          time: '2023-05-16T13:17:01.579Z',
        },
        {
          responseTime: 98,
          status: 'OK',
          time: '2023-05-16T13:33:07.271Z',
        },
        {
          responseTime: 6,
          status: 'OK',
          time: '2023-05-16T13:39:01.371Z',
        },
      ],
      percentages: [
        {
          value: 94.28572,
          status: 'OK',
        },
        {
          value: 5.714286,
          status: 'KO',
        },
      ],
    }
    return res(ctx.json(mockResponse))
  }),
]
