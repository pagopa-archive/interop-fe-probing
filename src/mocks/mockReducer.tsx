import { BrowserRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

const queryClient = new QueryClient()

function reducer(ui: any, { preloadedState, store, ...renderOptions }: any = {}) {
  function Wrapper({ children }: any) {
    return (
      <Router>
        <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
      </Router>
    )
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

export { reducer }
