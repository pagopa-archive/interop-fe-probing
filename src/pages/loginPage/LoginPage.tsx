import { LoginForm } from '../../components/forms/loginForm/LoginForm'
import { Grid, Typography, Link } from '@mui/material'
import { useTranslation } from 'react-i18next'

const LoginPage = () => {
  const { t } = useTranslation(['loginPage'])

  return (
    <Grid
      container
      direction="column"
      sx={{ height: '100%' }}
      justifyContent="center"
      rowGap={3}
      mt={5}
      mb={6}
    >
      <Grid item alignItems="center" justifyContent="center" sx={{ textAlign: 'center' }}>
        <Typography variant="h4" component="h1">
          {t('login', { ns: 'loginPage' })}
        </Typography>
      </Grid>
      <Grid item container alignSelf={'center'} width={'60%'} justifyContent={'center'}>
        <LoginForm />
      </Grid>
      <Grid item alignSelf={'center'}>
        <Typography>
          {t('forgotPassword1', { ns: 'loginPage' })}
          <Link href="" underline="none">
            {t('forgotPassword2', { ns: 'loginPage' })}
          </Link>
          {t('forgotPassword3', { ns: 'loginPage' })}
        </Typography>
      </Grid>
    </Grid>
  )
}
export default LoginPage
