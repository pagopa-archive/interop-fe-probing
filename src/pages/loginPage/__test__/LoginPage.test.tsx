import { describe, expect, test, vi } from 'vitest'
import LoginPage from '../LoginPage'
import { renderWithRouterAndQuery } from '../../../mocks/mockWrapper'
import { logRoles, screen, prettyDOM } from '@testing-library/react'
import * as auth from '../../../authetication/auth'
import userEvent from '@testing-library/user-event'
import * as router from 'react-router'

describe('LoginPage', () => {
  test('render component', () => {
    const { container } = renderWithRouterAndQuery(<LoginPage />)
    expect(container).toMatchSnapshot()
  })
})
