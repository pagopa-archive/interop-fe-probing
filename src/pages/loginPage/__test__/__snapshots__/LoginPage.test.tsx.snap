// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginPage > render component 1`] = `
<div>
  <main
    class="MuiStack-root css-1bw9v5o-MuiStack-root"
  >
    <h1
      class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
    >
      login
    </h1>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation16 MuiCard-root css-9cvwhb-MuiPaper-root-MuiCard-root"
    >
      <form>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-ljy4ha-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-1qy0967-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="username"
                  id="username-label"
                >
                  username
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="username"
                    required=""
                    type="email"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        username
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="password"
                  id="password-label"
                >
                  password
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="password"
                    required=""
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        password
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-131xi47-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-w4y956-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="submit"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-jb1ygz-MuiTypography-root"
              >
                 
                login
              </p>
            </button>
          </div>
        </div>
      </form>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
    >
      forgotPassword1
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-4m5fmp-MuiTypography-root-MuiLink-root"
        href="/recupero-password"
      >
        forgotPassword2
      </a>
      forgotPassword3
    </p>
  </main>
</div>
`;
